src/System.Security.Cryptography.X509Certificates/tests/ChainTests.cs

```diff
@@ -226,6 +226,142 @@ public static void BuildChainExtraStoreUntrustedRoot()
+            UntrustedIntermediateTrustedRoot,
+            TrustedIntermediateTrustedRoot,
+            MultipleCalls
+        }
+
+        [Theory]
+        [InlineData(false, X509ChainStatusFlags.UntrustedRoot, BuildChainCustomTrustStoreTestArguments.TrustedIntermediateUntrustedRoot)]
+        [InlineData(true, X509ChainStatusFlags.NoError, BuildChainCustomTrustStoreTestArguments.UntrustedIntermediateTrustedRoot)]
+        [InlineData(true, X509ChainStatusFlags.NoError, BuildChainCustomTrustStoreTestArguments.TrustedIntermediateTrustedRoot)]
+        [InlineData(true, X509ChainStatusFlags.NoError, BuildChainCustomTrustStoreTestArguments.MultipleCalls)]
+        public static void BuildChainCustomTrustStore(
+            bool chainBuildsSuccessfully,
+            X509ChainStatusFlags chainFlags,
+            BuildChainCustomTrustStoreTestArguments testArguments)
+        {
+            X509Certificate2 rootCert = null;
+            X509Certificate2 intermediateCert = null;
+
+            using (var microsoftDotCom = new X509Certificate2(TestData.MicrosoftDotComSslCertBytes))
+            using (var testCert = new X509Certificate2(Path.Combine("TestData", "test.pfx"), TestData.ChainPfxPassword))
+            using (var chainHolder = new ChainHolder())
+            {
+                X509Chain chain = chainHolder.Chain;
+                chain.ChainPolicy.RevocationMode = X509RevocationMode.NoCheck;
+                chain.ChainPolicy.VerificationTime = microsoftDotCom.NotBefore.AddSeconds(1);
+
+                chain.Build(microsoftDotCom);
+                rootCert = chain.ChainElements[2].Certificate;
+                intermediateCert = chain.ChainElements[1].Certificate;
+                chain.ChainPolicy.TrustMode = X509ChainTrustMode.CustomRootTrust;
+
+                switch (testArguments)
+                {
+                    case BuildChainCustomTrustStoreTestArguments.TrustedIntermediateUntrustedRoot:
+                        chain.ChainPolicy.ExtraStore.Add(rootCert);
+                        chain.ChainPolicy.CustomTrustStore.Add(intermediateCert);
+                        break;
+                    case BuildChainCustomTrustStoreTestArguments.UntrustedIntermediateTrustedRoot:
+                        chain.ChainPolicy.ExtraStore.Add(intermediateCert);
+                        chain.ChainPolicy.CustomTrustStore.Add(rootCert);
+                        break;
+                    case BuildChainCustomTrustStoreTestArguments.TrustedIntermediateTrustedRoot:
+                        chain.ChainPolicy.CustomTrustStore.Add(intermediateCert);
+                        chain.ChainPolicy.CustomTrustStore.Add(rootCert);
+                        break;
+                    case BuildChainCustomTrustStoreTestArguments.MultipleCalls:
+                        chain.ChainPolicy.CustomTrustStore.Add(intermediateCert);
+                        chain.ChainPolicy.CustomTrustStore.Add(rootCert);
+                        chain.Build(microsoftDotCom);
```

Comment:
chainHolder.DisposeChainElements(); after this build (so that the objects it created are disposed before the next chain.Build)